hw/ip_templates/ac_range_check/data/ac_range_check.hjson.tpl

```diff
@@ -122,12 +125,17 @@
       swaccess: "ro"
       hwaccess: "hwo"
       fields: [
-        { bits: "22:18"
+        { bits: "${14+nr_role_bits+nr_ctn_uid_bits+math.ceil(math.log(num_ranges, 2))-1}:${14+nr_role_bits+nr_ctn_uid_bits}"
```

Comment:
This is a bit of a mouthful. Could you do this by defining some variables at the start, based on the template parameters. Maybe something like this?
```python
denied_ctn_uid_lsb = 14 + nr_role_bits
deny_range_index_lsb = denied_ctn_uid_lsb + nr_ctn_uid_bits
deny_range_index_size = math.ceil(math.log(num_ranges, 2))
```